src/plugins/intel_gpu/src/graph/include/program_node.h

```diff
@@ -91,7 +91,7 @@ struct program_node {
                 if (u->get_dependencies().size() <= dep_idx) {
                     continue;
                 }
-                if (u->get_dependency(dep_idx).get_unique_id() == unique_id) {
```

Comment:
Please do not remove unique id usage in program_node. Its unnecessary for this task.

Reply:
rollback it.